File: openai-image-generation/node/generate-image.js
// @redocly-chunk {"steps": ["generate-image"]}
const response = await openai.images.generate({
 model: "dall-e-3",
 prompt: "a white siamese cat",
 n: 1,
 size: "/*@redocly-input {"id": "image-width"}*/x/*@redocly-input {"id": "image-height"}*/",
 @redocly-chunk {"steps": ["image-quality"]}
 quality: "hd"
 @redocly-chunk-end 
});
image_url = response.data[0].url;
// @redocly-chunk-end

File: openai-image-generation/python/generate-image.py
# @redocly-chunk {"steps": ["generate-image"]}
from openai import OpenAI
client = OpenAI()

response = client.images.generate(
  model="dall-e-3",
  prompt="a white siamese cat",
  size="/* @redocly-input {"id": "image-width"} */x/* @redocly-input {"id": "image-height"} */",
  # @redocly-chunk {"steps": ["image-quality"]}
  quality="hd"
  # @redocly-chunk-end
  n=1,
)

image_url = response.data[0].url
# @redocly-chunk-end